ipie/addons/adafqmc/utils/miscellaneous.py

```diff
@@ -0,0 +1,325 @@
+import torch
+import numpy as np
+
+# pylint: disable=import-error
+from pyscf import mcscf, lo
+from ipie.addons.adafqmc.hamiltonians.hamiltonian import HamObs
+from ipie.addons.adafqmc.utils.hf import hartree_fock
+from ipie.utils.from_pyscf import generate_integrals
+from ipie.utils.mpi import get_shared_array, have_shared_mem
+import h5py
+
+
+def generate_hamiltonian_from_pyscf(
```

Comment:
This is too much? Why does torch need to generate the integrals / inputs etc? Why not just cast arrays to torch from numpy or similar?

Reply:
Because the hamiltonian I used is a little bit different from ipie's, since I have to also include the perturbation part (e.g. dipole matrix), it is more convenient to write a new function to generate the hamiltonian in my pytorch definition, but I remember to have called the ipie function to generate intergrals.

Reply:
I think the solution would be to just call the ipie function and write a smaller function which is more self contained, and makes it obvious what is changing.

Reply:
ahh I remember the reason for this one now: I am generating frozen core hamiltonian here and the settings of the corresponding function in ipie is somewhat different from what I needed. That's why I wrote a separate function for this.